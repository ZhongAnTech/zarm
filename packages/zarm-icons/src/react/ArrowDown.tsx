import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowDown = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M494.094 733.426a41.472 41.472 0 01-20.023-11.126L140.16 388.388c-16.272-16.271-16.272-42.653 0-58.925 16.271-16.272 42.653-16.272 58.925 0L503.66 634.037 808.186 329.51c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L533.2 722.347c-10.607 10.607-25.508 14.3-39.106 11.08z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowDown);
export default ForwardRef;
