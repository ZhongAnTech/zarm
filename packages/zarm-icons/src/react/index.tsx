export { default as ArrowDown } from './ArrowDown';
export { default as ArrowLeft } from './ArrowLeft';
export { default as ArrowRight } from './ArrowRight';
export { default as ArrowUp } from './ArrowUp';
export { default as CloseCircleFill } from './CloseCircleFill';
export { default as CloseCircle } from './CloseCircle';
export { default as Close } from './Close';
export { default as DeleteKey } from './DeleteKey';
export { default as Empty } from './Empty';
export { default as Keyboard } from './Keyboard';
export { default as Minus } from './Minus';
export { default as PlusCircle } from './PlusCircle';
export { default as Plus } from './Plus';
export { default as Search } from './Search';
export { default as SuccessCircle } from './SuccessCircle';
export { default as Success } from './Success';
export { default as Volume } from './Volume';
export { default as WarningCircle } from './WarningCircle';
export { default as Warning } from './Warning';
