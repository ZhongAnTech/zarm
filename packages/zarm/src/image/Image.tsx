import React, { useState, useContext, createRef } from 'react';
import { createBEM } from '@zarm-design/bem';
import ImagePreview from '../image-preview';
import { ConfigContext } from '../n-config-provider';
import LazyDetector from './lazy-detector';

export const IMAGE_STATUS = {
  PENDING: 'pending',
  LOADING: 'loading',
  LOADED: 'loaded',
  FAILED: 'failed',
};

export interface ImageCssVars {
  '--background'?: React.CSSProperties['background'];
  '--text-color'?: React.CSSProperties['color'];
  '--font-size'?: React.CSSProperties['fontSize'];
  '--radius'?: React.CSSProperties['borderRadius'];
  '--width'?: React.CSSProperties['width'];
  '--height'?: React.CSSProperties['height'];
}

export interface ImageProps extends Omit<React.ImgHTMLAttributes<HTMLImageElement>, 'placeholder'> {
  placeholder?: React.ReactNode | boolean;
  fallback?: React.ReactNode | boolean;
  fit?: 'contain' | 'cover' | 'fill' | 'none' | 'scale-down';
  shape?: 'rect' | 'radius' | 'round' | 'circle';
  preview?: boolean;
  lazy?: boolean;
}

const Image = React.forwardRef<HTMLImageElement, ImageProps>((props, ref) => {
  const {
    src,
    placeholder,
    fallback,
    alt,
    onLoad,
    onError,
    onClick,
    className,
    width,
    height,
    style,
    fit,
    shape,
    preview,
    lazy,
    ...restProps
  } = props;
  const [status, setStatus] = useState(IMAGE_STATUS.LOADING);
  const [initialized, setInitialized] = useState(!lazy);

  const { prefixCls, locale } = useContext(ConfigContext);
  const bem = createBEM('image', { prefixCls });
  const cls = bem([{ [`${shape}`]: !!shape }, className]);

  const imgRef = ref || createRef<HTMLImageElement>();

  const inIframe = window.self !== window.top;

  const orientation = inIframe ? 'portrait' : undefined;

  const handleLoad = (event: React.SyntheticEvent<HTMLImageElement>) => {
    onLoad && onLoad(event);
    setStatus(IMAGE_STATUS.LOADED);
  };

  const handleError = (event: React.SyntheticEvent<HTMLImageElement>) => {
    onError && onError(event);
    setStatus(IMAGE_STATUS.FAILED);
  };

  const handleClick = (event: React.MouseEvent<HTMLImageElement, MouseEvent>) => {
    if (preview) {
      ImagePreview.show({ images: [src || ''], orientation });
    }
    onClick && onClick(event);
  };

  const renderImageStatus = {
    [IMAGE_STATUS.LOADED]: () => null,
    [IMAGE_STATUS.FAILED]: () => {
      if (fallback) {
        return (
          <div className={bem('fallback')}>
            {typeof fallback === 'boolean' ? locale?.Image?.loadFailed : fallback}
          </div>
        );
      }

      return null;
    },
    [IMAGE_STATUS.LOADING]: () => {
      if (placeholder) {
        return (
          <div className={bem('loading')}>
            {typeof placeholder === 'boolean' ? locale?.Image?.loading : placeholder}
          </div>
        );
      }

      return null;
    },
  };

  return (
    <div className={`${cls}`} style={{ width, height, ...style }}>
      {lazy && !initialized && (
        <LazyDetector
          onActive={() => {
            setInitialized(true);
          }}
        />
      )}
      <img
        className={bem('img')}
        src={initialized ? src : undefined}
        alt={alt}
        ref={imgRef}
        style={{
          objectFit: fit,
        }}
        {...restProps}
        onClick={handleClick}
        onError={handleError}
        onLoad={handleLoad}
      />
      {renderImageStatus[status]()}
    </div>
  );
});

Image.displayName = 'Image';
Image.defaultProps = {
  placeholder: false,
  fallback: false,
  fit: 'fill',
  shape: 'rect',
  preview: false,
  lazy: false,
};

export default Image;
