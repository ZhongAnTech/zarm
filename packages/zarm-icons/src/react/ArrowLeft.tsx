import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowLeft = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowLeft);
export default ForwardRef;
