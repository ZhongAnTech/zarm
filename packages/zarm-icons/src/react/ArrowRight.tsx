import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowRight = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowRight);
export default ForwardRef;
