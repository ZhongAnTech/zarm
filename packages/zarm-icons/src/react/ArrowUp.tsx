import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgArrowUp = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M513.176 285.05a41.472 41.472 0 0120.024 11.127L867.11 630.089c16.272 16.271 16.272 42.653 0 58.925s-42.654 16.272-58.925 0L503.612 384.44 199.085 688.967c-16.272 16.271-42.654 16.271-58.925 0-16.272-16.272-16.272-42.654 0-58.926L474.07 296.129c10.606-10.606 25.508-14.299 39.105-11.078z"
          fillRule="evenodd"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgArrowUp);
export default ForwardRef;
