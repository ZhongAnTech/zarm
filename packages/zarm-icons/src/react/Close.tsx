import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgClose = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <path
          d="M560.149 501.223l255.344 255.344c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L501.223 560.149 245.88 815.493c-16.272 16.272-42.654 16.272-58.925 0-16.272-16.272-16.272-42.654 0-58.926l255.344-255.344L186.954 245.88c-16.272-16.272-16.272-42.654 0-58.925 16.271-16.272 42.653-16.272 58.925 0l255.344 255.344 255.344-255.344c16.272-16.272 42.654-16.272 58.926 0 16.272 16.271 16.272 42.653 0 58.925L560.149 501.223z"
          fillRule="nonzero"
        />
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgClose);
export default ForwardRef;
