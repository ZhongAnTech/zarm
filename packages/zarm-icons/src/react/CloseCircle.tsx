import * as React from 'react';
import Icon from '../icon';
import type { IconProps } from '../icon';

const SvgCloseCircle = (props: IconProps, svgRef?: React.Ref<SVGSVGElement>) => {
  return React.createElement(Icon, {
    ...props,
    component: () => (
      <svg
        width="1em"
        height="1em"
        viewBox="0 0 1000 1000"
        fill="currentColor"
        focusable={false}
        aria-hidden="true"
        ref={svgRef}
      >
        <g fillRule="evenodd">
          <g>
            <path d="M500 460.716l137.493-137.493c10.848-10.848 28.436-10.848 39.284 0l.101.102c10.735 10.848 10.645 28.344-.203 39.08l-138.207 136.78 138.309 138.308c10.848 10.848 10.848 28.436 0 39.284-10.848 10.848-28.436 10.848-39.284 0L498.98 538.264 359.85 675.956c-10.41 10.303-27.188 10.26-37.545-.097-10.34-10.341-10.34-27.107 0-37.448L460.716 500 323.223 362.507c-10.848-10.848-10.848-28.436 0-39.284 10.848-10.848 28.436-10.848 39.284 0L500 460.716z" />
            <path
              d="M190.64 809.36c170.855 170.854 447.865 170.854 618.72 0 170.854-170.855 170.854-447.865 0-618.72-170.855-170.854-447.865-170.854-618.72 0-170.854 170.855-170.854 447.865 0 618.72zm-44.193 44.193c-195.263-195.262-195.263-511.844 0-707.106 195.262-195.263 511.844-195.263 707.106 0 195.263 195.262 195.263 511.844 0 707.106-195.262 195.263-511.844 195.263-707.106 0z"
              fillRule="nonzero"
            />
          </g>
        </g>
      </svg>
    ),
  });
};

const ForwardRef = React.forwardRef(SvgCloseCircle);
export default ForwardRef;
