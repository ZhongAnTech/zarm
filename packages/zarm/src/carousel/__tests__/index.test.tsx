import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import Image from '../../image';
import Carousel from '../index';

const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth');
const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight');
const IMGLIST = [
  'https://static.zhongan.com/website/health/zarm/images/banners/1.png',
  'https://static.zhongan.com/website/health/zarm/images/banners/2.png',
  'https://static.zhongan.com/website/health/zarm/images/banners/3.png',
];

const contentRender = () => {
  return IMGLIST.map((item, i) => {
    return (
      <div className="carousel__item__pic" key={+i}>
        <Image src={item} alt="" draggable={false} />
      </div>
    );
  });
};

const createCarousel = (props?, childrenLen = 3) => {
  const ITEMS = Array.from({ length: childrenLen }).map((_v, i) => i);
  return (
    <Carousel {...props}>
      {ITEMS.map((item, index) => {
        return <div key={+index}>{item}</div>;
      })}
    </Carousel>
  );
};

describe('Carousel', () => {
  it('base render correctly', () => {
    const wrapper = render(
      <div>
        {createCarousel()}
        {createCarousel({}, 0)}
        {createCarousel({ activeIndex: 1 }, 0)}
        {createCarousel({}, 1)}
      </div>,
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('style render correctly', () => {
    const style = { background: 'red' };
    const wrapper = render(createCarousel({ style }, 0));
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('prefixCls render correctly', () => {
    const prefixCls = 'za-test';
    const wrapper = render(createCarousel({ prefixCls }, 1));
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('className render correctly', () => {
    const className = 'za-wrapper-test';
    const wrapper = render(createCarousel({ className }, 0));
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('height render correctly', () => {
    const wrapper = render(createCarousel({ height: 150, direction: 'vertical' }));
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('pagination render correctly', () => {
    const wrapper = render(
      <div>
        {createCarousel({ showPagination: true })}
        {createCarousel({ showPagination: false })}
      </div>,
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  // it('activeIndex', () => {
  //   const wrapper = mount(createCarousel({ activeIndex: 1 }));
  //   expect(wrapper.state('activeIndex')).toEqual(1);
  // });

  it('swipeable', () => {
    const wrapper = render(createCarousel({ swipeable: false }));
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('autoPlay', () => {
    jest.useFakeTimers();
    const onChange = jest.fn();
    const animationDuration = 200;
    const autoPlayIntervalTime = 1000;
    const props = { autoPlay: true, animationDuration, autoPlayIntervalTime };
    render(createCarousel({ ...props, onChange }));

    act(() => {
      jest.advanceTimersByTime(autoPlayIntervalTime);
    });
    expect(onChange).toBeCalledWith(1);
  });

  beforeEach(() => {
    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 375 });
    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 200 });
  });
  afterAll(() => {
    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', originalOffsetWidth!);
    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', originalOffsetHeight!);
  });

  it('swipe change horizontal', () => {
    const onChange = jest.fn();
    const { container } = render(<Carousel onChange={onChange}>{contentRender()}</Carousel>);
    const element = container.querySelector('.za-carousel');
    fireEvent.mouseDown(element!, { pointerId: 1, clientX: 0, clientY: 0, buttons: 1 });
    fireEvent.mouseMove(element!, { pointerId: 1, clientX: -200, clientY: 0, buttons: 1 });
    fireEvent.mouseUp(element!, { pointerId: 1, clientX: -275 });
    expect(onChange).toBeCalledTimes(1);
  });

  it('swipe change vertical', () => {
    const onChange = jest.fn();
    const { container } = render(
      <Carousel onChange={onChange} direction="vertical" loop>
        {contentRender()}
      </Carousel>,
    );
    const element = container.querySelector('.za-carousel');
    fireEvent.mouseDown(element!, { pointerId: 1, clientX: 50, clientY: 0, buttons: 1 });
    fireEvent.mouseMove(element!, { pointerId: 1, clientX: 50, clientY: -100, buttons: 1 });
    fireEvent.mouseUp(element!, { pointerId: 1, clientX: 50, clientY: -350 });
    expect(onChange).toBeCalledTimes(1);
  });

  it('transitionend event', () => {
    const onChangeEnd = jest.fn();
    const { container } = render(createCarousel({ onChangeEnd }));
    fireEvent.click(container.getElementsByClassName('za-carousel__pagination__item')[1]);
    fireEvent.transitionEnd(container.getElementsByClassName('za-carousel__items')[0]);
    expect(onChangeEnd).toBeCalledWith(1);
  });
});
