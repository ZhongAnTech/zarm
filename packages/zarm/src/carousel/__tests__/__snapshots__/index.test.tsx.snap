// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel base render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="za-carousel za-carousel--horizontal"
    >
      <div
        class="za-carousel__items"
        style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
      >
        <div
          class="za-carousel__item"
        >
          0
        </div>
        <div
          class="za-carousel__item"
        >
          1
        </div>
        <div
          class="za-carousel__item"
        >
          2
        </div>
      </div>
      <div
        class="za-carousel__pagination"
      >
        <div
          class="za-carousel__pagination__item za-carousel__pagination__item--active"
        />
        <div
          class="za-carousel__pagination__item"
        />
        <div
          class="za-carousel__pagination__item"
        />
      </div>
    </div>
    <div
      class="za-carousel za-carousel--horizontal"
    >
      <div
        class="za-carousel__items"
        style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
      />
      <div
        class="za-carousel__pagination"
      />
    </div>
    <div
      class="za-carousel za-carousel--horizontal"
    >
      <div
        class="za-carousel__items"
        style="transition-duration: 0ms; transform: translate3d(-200px, 0px, 0);"
      />
      <div
        class="za-carousel__pagination"
      />
    </div>
    <div
      class="za-carousel za-carousel--horizontal"
    >
      <div
        class="za-carousel__items"
        style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
      >
        <div
          class="za-carousel__item"
        >
          0
        </div>
      </div>
      <div
        class="za-carousel__pagination"
      >
        <div
          class="za-carousel__pagination__item za-carousel__pagination__item--active"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Carousel className render correctly 1`] = `
<DocumentFragment>
  <div
    class="za-carousel za-carousel--horizontal za-wrapper-test"
  >
    <div
      class="za-carousel__items"
      style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
    />
    <div
      class="za-carousel__pagination"
    />
  </div>
</DocumentFragment>
`;

exports[`Carousel height render correctly 1`] = `
<DocumentFragment>
  <div
    class="za-carousel za-carousel--vertical"
  >
    <div
      class="za-carousel__items"
      style="height: 150px; transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
    >
      <div
        class="za-carousel__item"
      >
        0
      </div>
      <div
        class="za-carousel__item"
      >
        1
      </div>
      <div
        class="za-carousel__item"
      >
        2
      </div>
    </div>
    <div
      class="za-carousel__pagination"
    >
      <div
        class="za-carousel__pagination__item za-carousel__pagination__item--active"
      />
      <div
        class="za-carousel__pagination__item"
      />
      <div
        class="za-carousel__pagination__item"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Carousel pagination render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="za-carousel za-carousel--horizontal"
    >
      <div
        class="za-carousel__items"
        style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
      >
        <div
          class="za-carousel__item"
        >
          0
        </div>
        <div
          class="za-carousel__item"
        >
          1
        </div>
        <div
          class="za-carousel__item"
        >
          2
        </div>
      </div>
      <div
        class="za-carousel__pagination"
      >
        <div
          class="za-carousel__pagination__item za-carousel__pagination__item--active"
        />
        <div
          class="za-carousel__pagination__item"
        />
        <div
          class="za-carousel__pagination__item"
        />
      </div>
    </div>
    <div
      class="za-carousel za-carousel--horizontal"
    >
      <div
        class="za-carousel__items"
        style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
      >
        <div
          class="za-carousel__item"
        >
          0
        </div>
        <div
          class="za-carousel__item"
        >
          1
        </div>
        <div
          class="za-carousel__item"
        >
          2
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Carousel prefixCls render correctly 1`] = `
<DocumentFragment>
  <div
    class="za-carousel za-carousel--horizontal"
  >
    <div
      class="za-carousel__items"
      style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
    >
      <div
        class="za-carousel__item"
      >
        0
      </div>
    </div>
    <div
      class="za-carousel__pagination"
    >
      <div
        class="za-carousel__pagination__item za-carousel__pagination__item--active"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Carousel style render correctly 1`] = `
<DocumentFragment>
  <div
    class="za-carousel za-carousel--horizontal"
    style="background: red;"
  >
    <div
      class="za-carousel__items"
      style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
    />
    <div
      class="za-carousel__pagination"
    />
  </div>
</DocumentFragment>
`;

exports[`Carousel swipeable 1`] = `
<DocumentFragment>
  <div
    class="za-carousel za-carousel--horizontal"
  >
    <div
      class="za-carousel__items"
      style="transition-duration: 0ms; transform: translate3d(0px, 0px, 0);"
    >
      <div
        class="za-carousel__item"
      >
        0
      </div>
      <div
        class="za-carousel__item"
      >
        1
      </div>
      <div
        class="za-carousel__item"
      >
        2
      </div>
    </div>
    <div
      class="za-carousel__pagination"
    >
      <div
        class="za-carousel__pagination__item za-carousel__pagination__item--active"
      />
      <div
        class="za-carousel__pagination__item"
      />
      <div
        class="za-carousel__pagination__item"
      />
    </div>
  </div>
</DocumentFragment>
`;
